### ISSUE TITLE
Selected values does not work with a collection

### ISSUE THREAD

**Issue by AidasK** [LaravelCollective/html#259]
https://github.com/LaravelCollective/html/blob/5.3/src/FormBuilder.php#L714

Can be changed to:
```php
if (is_array($selected) or $selected instanceof \ArrayAccess) {
...
```

**Comment by vlakoff** [comment]
If I recall correctly `in_array()` doesn't work with `ArrayAccess`.

See code of `getCheckboxCheckedState()` method for a working way of supporting Collections.

**Comment by vlakoff** [comment]
So the code would be:
```php
    protected function getSelectedValue($value, $selected)
    {
        if (is_array($selected)) {
            return in_array($value, $selected) ? 'selected' : null;
        }
        elseif ($selected instanceof Collection) {
            return $selected->contains($value) ? 'selected' : null;
        }
        return ((string) $value == (string) $selected) ? 'selected' : null;
    }
```

**Comment by mlantz** [comment]
Hey @vlakoff and @AidasK is this still an issue? Should we add this code you have above?

**Comment by vlakoff** [comment]
I did some investigations:

* Currently, passing a Collection doesn't produce any errors, it silently matches nothing, in practice. (by the way, casting a Collection to string produces its JSON representation. That's kinda clever, though too magic imho)
* I noticed a little flaw. `placeholderOption` calls `getSelectedValue(null, $selected)` then, because of the loose comparison, `null` would match the `0` in e.g. `in_array([0, 1, 2])`. Though, `Collection->contains` would have the exact same flaw. So at least it's consistent :)

Finally, should we add the above code? I'm still a bit undecided, but I'd bend towards yes.
